dragonfly_doe2/doe/shades.py

```diff
@@ -0,0 +1,118 @@
+            face.vertices[0], face.vertices[2]).length
+        face_len_two = LineSegment3D.from_end_points(
+            face.vertices[1], face.vertices[3]).length
+
+        face = face if isclose(face_len_one, face_len_two) else \
+            Face3D.from_regular_polygon(face.boundary_polygon2d)
+
+        face = face if face.normal.z <= 0 else face.flip()
+        # todo: add isclose check for geometry
+        shade_type = "FIXED-SHADE" if shade_type == None else shade_type
+        orig_point = face.lower_left_corner
+        x_ref, y_ref, z_ref = orig_point[0], orig_point[1], orig_point[2]
+
+        height = abs(round(LineSegment3D.from_end_points(
+            face.upper_right_corner, face.lower_right_corner).length, 4))
+        # ? I don't like the lack of uniformity between the height and width methods; but it DOES work..
+        width = face.plane.xyz_to_xy(face.max) - face.plane.xyz_to_xy(face.min)
+        width = abs(round(width[0], 4))
+
+        tilt = round(degrees(face.normal.angle(Vector3D(0, 0, 1))), 0)
+        y_axis = Vector3D(0, 1, 0)
+        projected_normal = Vector3D(face.normal[0], face.normal[1], 0)
+        try:
+            azimuth = math.degrees(projected_normal.angle(y_axis))
+            cross = face.normal.cross(y_axis)
+            if cross.z < 0:
+                azimuth = 360 - azimuth
+        except ZeroDivisionError:
+            # horizontal
+            azimuth = 180 if face.normal.z > 0 else -180
+
+        return cls(
+            name=indx_identifier, shade_type=shade_type, height=height, width=width,
+            x_ref=x_ref, y_ref=y_ref, z_ref=z_ref, azimuth=azimuth, tilt=tilt)
+
+    def to_inp(self):
+        """Returns *.inp shade object string"""
+        return f'{self.name}   = {self.shade_type}\n   ' \
+            f'HEIGHT           = {self.height}\n   ' \
+            f'WIDTH            = {self.width}\n   ' \
+            f'TRANSMITTANCE    = {self.transmittance}\n   ' \
+            f'X-REF            = {self.x_ref}\n   ' \
+            f'Y-REF            = {self.y_ref}\n   ' \
+            f'Z-REF            = {self.z_ref}\n   ' \
+            f'AZIMUTH          = {self.azimuth}\n   ' \
+            f'TILT             = {self.tilt}\n   ..'
+
+    def __repr__(self):
+        return self.to_inp()
+
+
+@dataclass
+class Doe2ShadeCollection:
+
+    doe_shades: List[Doe2Shade]
+
+    @classmethod
+    def from_df_context_shades(cls, df_shades: [ContextShade]):
+        """Generate doe2 fixed shades from dragonfly context shades"""
+        shade_faces = []
+
+        for shade in df_shades:
+            for i, geom in enumerate(shade.geometry):
+                shade_faces.append((geom,
+                                    short_name(f'{shade.identifier}_{i}', 32)))
+        doe_shades = [Doe2Shade.from_face3d(obj[0], obj[1]) for obj in shade_faces]
+
+        return cls(doe_shades=doe_shades)
+
+    def to_inp(self):
+
+        block = [fb.fixBldgShade]
```

Comment:
Maybe it's a good time to rename this too?

```suggestion
        block = [fb.fix_bldg_shade]
```